tests/suites/test_suite_psa_crypto.function

```diff
@@ -9591,6 +9591,77 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void derive_key_custom(int alg_arg,
```

Comment:
I decided to duplicate the code (which I usually try hard not to do) for several reasons. The duplication isn't much of a problem because it's test code, and it's is only going into an LTS branch so the duplicated code is unlikely to change much. The 4.0 version won't have the legacy functions so it won't have the legacy tests either. Conversely, due to the different interfaces, the test cases need to be slightly different, and the code invoking the functions needs to be slightly different. We could use a couple of auxiliary functions to avoid repeating about 10 lines of pretty straightforward code, but I don't think it's worth it.

Reply:
> test_suite_psa_crypto.function which is already far above 10k lines.

[I know](https://github.com/Mbed-TLS/mbedtls/issues/4967) but that's out of scope here.